cloudify_marketplace/blueprint/scripts/configure.py

```diff
@@ -33,16 +52,251 @@ def regenerate_host_keys():
+
+    # TODO: Make this a script that runs every minute called by manager:
+    # If <file> exists, json.load file
+    # while <file>.execution_id != execution finished:
+    #   sleep(0.5)
+    # <apply changes>
+
+    # new_certs is expected to be a list of tuples with
+    # [(<new cert path>, <path to original cert (to replace)>), ...]
+    restart_file_path = '/tmp/cloudify_configuration_restart_services'
+    with open(restart_file_path, 'w') as restart_file_handle:
+        restart_file_handle.write('#! /usr/bin/env bash\n')
+
+        # Stop the services
+        for service in services:
+            restart_file_handle.write('service %s stop\n' % service)
+
+        # Copy certs to their destinations
+        for cert, destination in new_certs:
+            restart_file_handle.write('cp {cert} {destination}'.format(
+                cert=cert,
+                destination=destination,
+            ))
+
+        # Start the services again
+        for service in services:
+            restart_file_handle.write('service %s start\n' % service)
+
+        # Clean up
+        for cert, _ in new_certs:
+            restart_file_handle.write('rm -f {cert}'.format(
+                cert=cert,
+            ))
+        #restart_file_handle.write('rm %s\n' % restart_file_path)
+    subprocess.call([
+        'sudo',
+        'at',
+        'now', '+', '1', 'minutes',
+        '-f', restart_file_path,
+    ])
+
+
+def regenerate_broker_certificates(subjectaltnames):
+    subjectaltnames = get_general_subjectaltnames(subjectaltnames)
+
+    private_cert_path = '/etc/rabbitmq/rabbit-priv.pem'
+    public_cert_path = '/etc/rabbitmq/rabbit-pub.pem'
+    tmp_private = '/tmp/broker-private'
+    tmp_public = '/tmp/broker-public'
+    build_certs(
+        private_key_path=tmp_private,
+        public_key_path=tmp_public,
+        subjectaltnames=subjectaltnames,
+    )
+
+    new_certs = [
+        (tmp_private, private_cert_path),
+        (tmp_public, public_cert_path),
+    ]
+
+    for cert_path in [
+        '/opt/manager/amqp_pub.pem',
+        '/opt/mgmtworker/amqp_pub.pem',
+        '/opt/amqpinflux/amqp_pub.pem',
+    ]:
+        new_certs.append((tmp_public, cert_path))
+
+    return new_certs
+
+
 def main():
     regenerate_host_keys()
 
     authorize_user_ssh_key(inputs['user_ssh_key'])
 
+    security_enabled = os.path.exists(
+        '/root/.cloudify_image_security_enabled'
+    )
+    if security_enabled:
+        new_certs = []
+
+        certs = regenerate_broker_certificates(
+            inputs['broker_names_and_ips']
+        )
+        new_certs.extend(certs)
+        certs = regenerate_manager_certificates(
+            inputs['manager_names_and_ips']
+        )
+        new_certs.extend(certs)
+
+        replacement_data = {
+            'execution_id': ctx.execution_id,
+            # Get these from somewhere, somehow...
+            'cloudify_username': 'cloudify',
```

Comment:
Yeah, I checked as well and couldn't find it. We need to figure whether to retrieve them or to enforce the initial default for the manager creds. I'd prefer retrieving them.